### PR TITLE
Avoid calling `numpy.nonzero` on 0-dim arrays

### PR DESCRIPTION
**Pull request by asi1024**
Part of #2370. 

According to [NumPy 1.17.0 release notes](https://github.com/numpy/numpy/releases/tag/v1.17.0), calling nonzero on 0d arrays is deprecated. I fixed the following three test failures.
- `tests.cupy_tests.sorting_tests.test_search.TestNonzero`
  - Made `cupy.nonzero` raise DeprecationWarning when `ndim` is `0`.
- `tests.cupy_tests.testing_tests.test_helper.TestIgnoreOfNegativeValueDifferenceOnCpuAndGpu`, `tests.cupy_tests.linalg_tests.test_einsum.TestEinSumUnaryOperation`
  - The functions wrapped by `numpy_cupy_allclose` calls `numpy.nonzero` internally. The messages in `DeprecationWarning` suggested "Use `atleast_1d(cond).nonzero()` if the old behavior was intended.", but I didn't so because I found better implementation.

### CODE REVIEW

File: tests/cupy_tests/linalg_tests/test_einsum.py [hunk 1]
```diff
@@ -269,9 +269,6 @@ def test_too_many_dims3(self, xp):
     {'shape_a': (2, 3), 'subscripts': 'ij->...ij'},  # do nothing
     {'shape_a': (2, 3), 'subscripts': 'ij...->ij'},  # do nothing
     {'shape_a': (2, 3), 'subscripts': 'i...j->ij'},  # do nothing
-
-    {'shape_a': (), 'subscripts': ''},  # do nothing
-    {'shape_a': (), 'subscripts': '->'},  # do nothing
```
Comment (niboshi) [inline tests/cupy_tests/linalg_tests/test_einsum.py]: I think these patterns should be supported.

```
>>> numpy.__version__
'1.17.0'
>>> numpy.einsum('', numpy.ones(()))
1.0
>>> numpy.einsum('->', numpy.ones(()))
1.0
```

Comment (asi1024) [reply tests/cupy_tests/linalg_tests/test_einsum.py]: The test failure is due to `numpy_cupy_allclose` `calling numpy.nonzero` and these patterns should be supported as you say. Thanks.

File: tests/cupy_tests/sorting_tests/test_search.py [hunk 2]
```diff
@@ -221,6 +219,21 @@ def test_nonzero(self, xp, dtype):
         return xp.nonzero(array)
 
 
+@testing.parameterize(
+    {'array': numpy.empty(0)},
+    {'array': numpy.empty(1)},
```
Comment (niboshi) [inline tests/cupy_tests/sorting_tests/test_search.py]: Should be `numpy.array`. `numpy.empty(0)` and `numpy.empty(1)` result in `ndim=1`.

Comment (asi1024) [reply tests/cupy_tests/sorting_tests/test_search.py]: Fixed.

**Comment by niboshi** [comment]
jenkins test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**